backends/dpdk/dpdkArch.cpp

```diff
@@ -467,6 +467,215 @@ bool CollectMetadataHeaderInfo::preorder(const IR::Type_Struct *s) {
+                                        bit<13> fragOffset;
+                                        ...
+                                   }  
+                                   is converted into 
+                                   
+                                   header ipv4_t {
+                                        ...
+                                        bit<16>  flags_fragOffset;
+                                        ...
+                                   }
+                                    Here, for "flags", information saved are :
+                                        ModifiedName = flags_fragOffset;
+                                        header str   =  ipv4_t; [This is used if multiple headers 
+                                                                 have field with same name]
+                                        width        = 16;
+                                        offset       = 3;
+                                        lsb          = 3;
+                                        msb          = 15;
+                        */
+                        for (auto s = field_name_list.begin(); s != field_name_list.end(); s++) {
+                            hdrFieldInfo fieldObj;
+                            fieldObj.modifiedName = modifiedName;
+                            fieldObj.headerStr = st->name.name;
+                            fieldObj.modifiedWidth = size_sum_so_far;
+                            fieldObj.fieldWidth = s->second.fieldWidth;
+                            fieldObj.lsb = offset;
+                            fieldObj.msb = offset + s->second.fieldWidth - 1;
+                            fieldObj.offset = offset;
+                            structure->hdrFieldInfoList[s->first].push_back(fieldObj);
+                            offset += s->second.fieldWidth;
+                        }
+                        fields->push_back(new IR::StructField(IR::ID(modifiedName),
+                                          IR::Type_Bits::get(size_sum_so_far)));
+                        size_sum_so_far = 0;
+                        modifiedName = "";
+                        field_name_list.clear();
+                    }
+                } else {
+                    ::error("8-bit Alignment for Header Structure '%1%' is not possible",
+                            st->name.name);
+                    return st;
+                }
+            } else {
+                fields->push_back(field);
+            }
+        }
+        /* Throw error if there is non-aligned field present in header */
+        if (size_sum_so_far != 0) {
+            ::error("8-bit Alignment for Header Structure '%1%' is not possible", st->name.name);
+            return st;
+        }
+        return new IR::Type_Header(IR::ID(st->name), st->annotations, *fields);
+    }
+
+    return st;
+}
+
+/* This function process the occurance of non-aligned fields and replaces it with
```

Comment:
occurrence*

Reply:
I will address this while raising review in open source